actors/verifreg/src/lib.rs

```diff
@@ -1017,13 +1010,10 @@ fn validate_claim_extension(
 }
 
 // Checks that an address corresponsds to a miner actor.
-fn resolve_miner_id(rt: &mut impl Runtime, addr: &Address) -> Result<ActorID, ActorError> {
-    let id = rt.resolve_address(addr).with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
-        format!("failed to resolve provider address {}", addr)
-    })?;
+fn resolve_miner_id(rt: &mut impl Runtime, id: &ActorID) -> Result<ActorID, ActorError> {
```

Comment:
Please rename this to `check_miner_id`, since this no longer does resolution.
Change the parameter to `ActorID`, not `&ActorID`.
Change the return type to `Result<(), ActorError>` since the return is just the parameter, and at the call site you can use `req.provider` instead of `provider_id`.